From 7d41f29d93c14d2a4e23e616d30962c3c52af3cd Mon Sep 17 00:00:00 2001
From: Yan Wang <yan.wang@linux.intel.com>
Date: Mon, 26 Mar 2018 15:02:42 +0800
Subject: [PATCH 1359/2345] ASoC: APL/CNL: Fix of apl_trace_init().

Add one check to output error log if no available capture stream for
DMA trace.
And remove previous TODO comments of only the first capture stream
available because it has been fixed by the patch on SOF.

Signed-off-by: Yan Wang <yan.wang@linux.intel.com>
---
 sound/soc/sof/hw-apl.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/sound/soc/sof/hw-apl.c b/sound/soc/sof/hw-apl.c
index 9584be6..38090dd 100644
--- a/sound/soc/sof/hw-apl.c
+++ b/sound/soc/sof/hw-apl.c
@@ -887,9 +887,6 @@ static int apl_trace_init(struct snd_sof_dev *sdev, u32 *stream_tag)
 	/*
 	 * iterate capture stream array and find the first available one for
 	 * DMA trace transferring.
-	 * TODO: Currently only the first capture stream is available. In the
-	 * future, DMA trace should use the last one capture stream for
-	 * avoiding conflict with normal capture stream.
 	 */
 	for (i = 0; i < hdev->num_capture; i++) {
 		if (!hdev->cstream[i].open) {
@@ -900,7 +897,14 @@ static int apl_trace_init(struct snd_sof_dev *sdev, u32 *stream_tag)
 		}
 	}
 
-	/* initialize capture stream, set BDL address and return corresponding
+	if (!sdev->dtrace_stream) {
+		dev_err(sdev->dev,
+			"error: no available capture stream for DMA trace\n");
+		return -ENODEV;
+	}
+
+	/*
+	 * initialize capture stream, set BDL address and return corresponding
 	 * stream tag which will be sent to the firmware by IPC message.
 	 */
 	return apl_trace_prepare(sdev);
-- 
2.7.4

